crates/matrix-sdk/src/event_cache/room/mod.rs

```diff
@@ -530,30 +533,39 @@ mod private {
             let store_lock = store.lock().await?;
 
             let linked_chunk_id = LinkedChunkId::Room(&room_id);
-            let linked_chunk = match store_lock
+
+            // Load the full linked chunk, so as to feed the order tracker.
+            //
+            // If loading the full linked chunk failed, we'll clear the event cache, as it
+            // indicates that at some point, there's some malformed data.
+            let fully_loaded_linked_chunk =
+                match Self::load_full_linked_chunk(&*store_lock, linked_chunk_id).await {
```

Comment:
It seems we are loading all chunks with all their items. It seems to be quite annoying and quite a blocker to me. It will impact performance too much. At best we must only load the chunks **without** their items.